src/scripts/components/layers/time-playback/time-playback.ts

```diff
@@ -13,27 +16,60 @@ interface Props {
   maxTime: number;
   speed?: number;
   step?: number;
+  mainLayerId?: string | null;
+  compareLayerId?: string | null;
 }
 
 const TimePlayback: FunctionComponent<Props> = ({
   minTime,
   maxTime,
   speed = PLAYBACK_SPEED,
-  step = PLAYBACK_STEP
+  step = PLAYBACK_STEP,
+  mainLayerId,
+  compareLayerId
 }) => {
   const dispatch = useDispatch();
   const time = useSelector(timeSelector);
 
+  const [nextTime, setNextTime] = useState(time);
+
   useInterval(() => {
     let newTime = time + step;
 
     if (newTime > maxTime) {
       newTime = minTime;
     }
 
-    dispatch(setGlobeTime(newTime));
+    // don't immediately set the new time, since we might have to wait
+    // for the layer to complete loading the previous timestamp.
+    setNextTime(newTime);
   }, speed);
 
+  // before the globe-time is updated, make sure the layers have actually
+  // rendered the current time.
+
+  // get the state for both possible layers. When no layer is specified, we
+  //   use 'idle' as to not interfere with the loading logic
+  const layerLoadingState = useSelector(layerLoadingStateSelector);
+  const mainLayerState = mainLayerId
+    ? layerLoadingState[mainLayerId]
+    : ('idle' as LayerLoadingState);
+  const compareLayerState = compareLayerId
+    ? layerLoadingState[compareLayerId]
+    : ('idle' as LayerLoadingState);
```

Comment:
maybe we should move this into the parent component and pass it here as a single prop `layersLoaded`? I think that feels slightly more correct than to pass the layer-ids into this component.